api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupANCRegister.java

```diff
@@ -198,43 +193,56 @@ private DataSetDefinition dataSetDefinition() {
         //start constructing of the dataset
         PersonAttributeType attribute = Context.getPersonService().getPersonAttributeTypeByUuid("14d4f066-15f5-102d-96e4-000c29c2a5d7");
```

Comment:
@carapai take time and study [this](https://github.com/METS-Programme/openmrs-module-ugandaemr-reports/blob/master/api/src/main/java/org/openmrs/module/ugandaemrreports/definition/data/definition/CalculationDataDefinition.java). All calculations are evaluated and paramaters passed from the evaluation context. Here we base on **onDate** which is mapped to endDate. Note how that parameter is evaluated [here](https://github.com/METS-Programme/openmrs-module-ugandaemr-reports/blob/master/api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupANCRegister.java#L209) and passed [here](https://github.com/METS-Programme/openmrs-module-ugandaemr-reports/blob/master/api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupANCRegister.java#L149-L155)